tools/jil/driver/TestRun.js

```diff
@@ -107,35 +108,24 @@ class TestRun extends EventEmitter {
 
     harness.addTest(testName, opts, (t) => {
       let startTime = Date.now()
-      let ended = false
 
       harness.pause()
+      observeTapeTest(t, onTestFinished, onTestResult)
```

Comment:
This covers the same tests as before, I just extracted the logic that was here into the observeTapeTest function that is in a separate module now.

There is probably an opportunity to consolidate some of the logic into less classes. I think it might make sense to move this observeTapeTest logic to the harness, since that is what deals with running the tape tests. That would be a bigger refactor, I think. I tried to keep the change smaller for now.